### ISSUE TITLE
Test with ASAN_OPTIONS=detect_stack_use_after_return=1

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#829]
On CI, when we run with ASAN, enable the `detect_stack_use_after_return=1` `ASAN_OPTIONS` flag to catch more bugs.

**Comment by AlekLefebvre** [comment]
#863 I submitted this PR.

I also tested the flags suggested on the [AddressSanitizer FAQ] for more aggressive diagnostics (https://github.com/google/sanitizers/wiki/AddressSanitizer#faq) and they are not triggering new errors on this project. Would you like to enable them too?

```
CFLAGS += -fsanitize-address-use-after-scope
ASAN_OPTIONS=strict_string_checks=1:detect_stack_use_after_return=1:check_initialization_order=1:strict_init_order=1
```

**Comment by strager** [comment]
> Would you like to enable them too?

Sure!

**Comment by strager** [comment]
@AlekLefebvre did this in PRs #863 and #864 (commits 4d41ecb465f5649a20954047d0d4f6096981e78b and e6716d5bf7c050b0915d6baefc87718dfe2cd7c3).
